pkg/services/accesscontrol/eval/all_test.go

```diff
@@ -0,0 +1,109 @@
+package eval
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAll_Evaluate(t *testing.T) {
+	tests := []evaluateTestCase{
+		{
+			desc: "should return true for one that matches",
+			evaluator: All(
+				Permission("settings:write", Combine("settings", ScopeAll)),
+			),
+			permissions: map[string]map[string]struct{}{
+				"settings:write": {"settings:**": struct{}{}},
+			},
+			expected: true,
+		},
+		{
+			desc: "should return true for several that matches",
+			evaluator: All(
+				Permission("settings:write", Combine("settings", ScopeAll)),
+				Permission("settings:read", Combine("settings", "auth.saml", ScopeAll)),
+			),
+			permissions: map[string]map[string]struct{}{
+				"settings:write": {"settings:**": struct{}{}},
+				"settings:read":  {"settings:**": struct{}{}},
+			},
+			expected: true,
+		},
+		{
+			desc: "should return false if one does not match",
+			evaluator: All(
+				Permission("settings:write", Combine("settings", ScopeAll)),
+				Permission("settings:read", Combine("settings", "auth.saml", ScopeAll)),
+				Permission("report:read", Combine("reports", ScopeAll)),
+			),
+			permissions: map[string]map[string]struct{}{
+				"settings:write": {"settings:**": struct{}{}},
+				"settings:read":  {"settings:**": struct{}{}},
+				"report:read":    {"report:1": struct{}{}},
+			},
+			expected: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			ok, err := test.evaluator.Evaluate(test.permissions)
+			assert.NoError(t, err)
+			assert.Equal(t, test.expected, ok)
+		})
+	}
+}
+
+func TestAll_Inject(t *testing.T) {
+	tests := []injectTestCase{
+		{
+			desc:     "should inject correct param",
+			expected: true,
+			evaluator: All(
+				Permission("reports:read", Combine("reports", Parameter(":reportId"))),
+				Permission("settings:read", Combine("settings", Parameter(":settingsId"))),
+			),
+			params: map[string]string{
+				":id":         "10",
```

Comment:
I think this is unused there, but it might be on purpose? To show we select the correct parameters?
```suggestion
```

Reply:
Same in the any_test.go